build.sbt

```diff
@@ -122,82 +104,45 @@ def jarsAndListGenerator = Def.taskDyn {
       (project, jarFiles, loadedJarFiles)
     }
 
-  // Add the built-in benchmarks.
-  val coreJarTask = Def.task {
-    val coreJar = (packageBin in (renaissanceCore, Runtime)).value
-    val coreJarDest =
-      (resourceManaged in Compile).value / "benchmarks" / "core" / coreJar.getName
-    coreJarDest.getParentFile.mkdirs()
-    Files.copy(coreJar.toPath, coreJarDest.toPath, StandardCopyOption.REPLACE_EXISTING)
-    ("benchmarks/core", Seq(coreJarDest), Seq(coreJarDest))
-  }
-  val jarTasks = coreJarTask +: projectJarTasks
   val nonGpl = nonGplOnly.value
 
-  // Flatten list, create a groups-jars file, and a benchmark-group file.
-  flattenTasks(jarTasks).map { groupJars =>
-    val jarListFile = (resourceManaged in Compile).value / "groups-jars.txt"
+  // Flatten list and create a groups-jars file and the benchmark-details file.
+  flattenTasks(projectJarTasks).map { groupJars =>
+    // Add the benchmarks from the different project groups.
     val jarListContent = new StringBuilder
-    val benchGroupFile = (resourceManaged in Compile).value / "benchmark-group.txt"
-    val benchGroupContent = new StringBuilder
-    val benchDetailsFile = (resourceManaged in Compile).value / "benchmark-details.properties"
-    val benchDetailsStream = new java.io.FileOutputStream(benchDetailsFile)
     val benchDetails = new java.util.Properties
 
-    // Add the benchmarks from the different project groups.
     for ((project, allJars, loadedJars) <- groupJars) {
-      val jarLine = loadedJars
-        .map(jar => project + "/" + jar.getName)
-        .mkString(",")
+      val jarLine = loadedJars.map(jar => project + "/" + jar.getName).mkString(",")
       val projectShort = project.stripPrefix("benchmarks/")
       jarListContent.append(projectShort).append("=").append(jarLine).append("\n")
-      for ((name, distroLicense, licenses, description, repetitions) <- listBenchmarks(
-             project,
-             allJars
-           )) {
-        if (!nonGpl || distroLicense == "MIT") {
-          benchGroupContent.append(name).append("=").append(projectShort).append("\n")
-          benchDetails.setProperty("benchmark." + name + ".description", description)
-          benchDetails.setProperty("benchmark." + name + ".repetitions", repetitions.toString)
-          benchDetails.setProperty("benchmark." + name + ".distro", distroLicense.toString)
-          benchDetails.setProperty("benchmark." + name + ".licenses", licenses.toString)
+
+      // Scan project jars for benchmarks and fill the property file.
+      for (benchInfo <- listBenchmarks(project, allJars)) {
+        if (!nonGpl || benchInfo.distro() == License.MIT) {
```

Comment:
As a side, at some point we should replace `nonGpl` with `isGpl`, I think. Easier when you don't have to think with double negations.

Reply:
Yes, I was peering at the quizzically as well :-)